set.mm

```diff
@@ -730651,6 +730957,1157 @@ last digit of the integer part (for ` b = ` 10 the first digit before
+      UJZUIZAEDWKWLADELUKULZUMZACDACIUJZWLUIZTZWIUNWMWNUOAWBWGWAAWBVPVSRQZVPGRQ
+      ZSQZWCEGRQZDGRQZPQZVODRQZPQZSQZWGAVPVSGWMWRWIUPAXAWCXEPQZXDSQXGAWSXHWTXDS
+      AVPVQRQZVRRQVOVRRQWSXHAXIVOVRRAVOVPWJWMWNUQURAVPVQVRWMWOWQUSAVOCDWJWPWLVA
+      VBAEDGWKWLWIVCVDAWCXEXDAVOCWJWPTAVODWJWLTAXBXCAEGWKWITADGWLWITUIVEVFAXFWF
+      WCSAXFWDGDRQZPQZHDRQZXJPQZPQWDXLPQWFAXDXKXEXMPAXBWDXCXJPAEGWKWIVGADGWLWIV
+      GVDAHGDWHWIWLVCVDAWDXLXJAGEWIWKTAHDWHWLTAGDWIWLTVJAXLWEWDPAHDWHWLVGVHVIVH
+      VIVKVL $.
+  $}
+
+  ${
+    affineid.f $e |- ( ph -> A e. CC ) $.
+    affineid.x $e |- ( ph -> T e. CC ) $.
+    $( Identity of an affine combination.  (Contributed by AV, 2-Feb-2023.) $)
+    affineid $p |- ( ph -> ( ( ( 1 - T ) x. A ) + ( T x. A ) ) = A ) $=
+      ( c1 cmin co cmul caddc 1cnd subdird mulid2d oveq1d eqtrd mulcld npcand )
+      AFCGHBIHZCBIHZJHBSGHZSJHBARTSJARFBIHZSGHTAFCBAKEDLAUABSGABDMNONABSDACBEDP
+      QO $.
+  $}
+
+  ${
+    resum2sqcl.q $e |- Q = ( ( A ^ 2 ) + ( B ^ 2 ) ) $.
+    $( The sum of two squares of real numbers is a real number.  (Contributed
+       by AV, 7-Feb-2023.) $)
+    resum2sqcl $p |- ( ( A e. RR /\ B e. RR ) -> Q e. RR ) $=
+      ( cr wcel wa c2 cexp co caddc simpl resqcld simpr readdcld syl5eqel ) AEF
+      ZBEFZGZCAHIJZBHIJZKJEDSTUASAQRLMSBQRNMOP $.
+
+    $( The sum of the square of a nonzero real number and the square of another
+       real number is greater than zero.  (Contributed by AV, 7-Feb-2023.) $)
+    resum2sqgt0 $p |- ( ( ( A e. RR /\ A =/= 0 ) /\ B e. RR ) -> 0 < Q ) $=
+      ( cr wcel cc0 wne wa c2 co caddc clt simpl resqcld adantr simpr wbr sqgt0
+      cexp cle sqge0 adantl addgtge0d syl6breqr ) AEFZAGHZIZBEFZIZGAJTKZBJTKZLK
+      CMUJUKULUHUKEFUIUHAUFUGNOPUJBUHUIQOUHGUKMRUIASPUIGULUARUHBUBUCUDDUE $.
+  $}
+
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  Spheres and lines in real Euclidean spaces
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  $c LineM Sphere $.
+
+  $( Declare the syntax for lines in generalized real Euclidean spaces. $)
+  cline $a class LineM $.
+
+  $( Declare the syntax for spheres in generalized real Euclidean spaces. $)
+  csph $a class Sphere $.
+
+  ${
+    $d p t w x y $.
+    $( Definition of lines passing through two different points in a left
+       module (or any extended structure having a base set, an addition, and a
+       scalar multiplication).  (Contributed by AV, 14-Jan-2023.) $)
+    df-line $a |- LineM = ( w e. _V |-> ( x e. ( Base ` w ) ,
+                                         y e. ( ( Base ` w ) \ { x } )
+                    |-> { p e. ( Base ` w ) | E. t e. ( Base ` ( Scalar ` w ) )
+                          p = ( ( ( ( 1r ` ( Scalar ` w ) )
+                                    ( -g ` ( Scalar ` w ) ) t ) ( .s ` w ) x )
+                                ( +g ` w ) ( t ( .s ` w ) y ) ) } ) ) $.
```

Comment:
This definition is very similar to the `LineG` slot of [``( toTG ` w)``](https://us.metamath.org/mpeuni/ttgval.html). Is the only difference the fact that it uses a generic scalar, while `toTG` assume that the scalars are reals?

It appears that just a few theorems down the line, you are switching back to real scalars only.

Reply:
You are right, there is a relatinship between both concepts. The main difference is that LineG is a slot only (which may contain anything), whereas LineM defines lines themselves. Another important difference is that LineM is based on properties of left modules, whereas LineG is based on betweenness. Nevertheless, looking at theorem ~elntg, something like `` ( LineG ` ( EEG ` N ) ) = ( LineM ` ( EEhil ` N ) ) `` or `` ( LineG ` ( toTG ` ( EEhil ` N ) ) ) = ( LineM ` ( EEhil ` N ) )`` should be provable.
Then the definition of LineM could have been (or is equivalet to) `` LineM = ( w e. _V |-> ( LineG ` ( toTG ` w ) )``.

Reply:
Yes, that's exactly the spirit of my remark. I find it particularly satisfying to build such bridges between different parts of `set.mm`, and to reach an economy of symbols and definitions.

I think we can commit like this, and maybe simplify, or add these "bridge" proofs later?

Reply:
With my latest commit I provide a theorem showing ``( LineG ` ( EEG ` N ) ) = ( LineM ` ( EEhil ` N ) )``. It required a lot of effort to provide its proof, but the simplicty of the result was worth it.

Reply:
Yes, it was actually much harder than I thought! Thanks for this very interesting addition.